### ISSUE TITLE
There is a statement that I don’t understand in "Collections are smart pointers"

### ISSUE THREAD

**Issue by kaifoon** [rust-unofficial/patterns#257]
Disadvantages
Methods and traits only available via dereferencing are not taken into account when bounds checking, so generic programming with data structures using this pattern can get complex (see the Borrow and AsRef traits, etc.).
why  method and traits  only available via dereferencing are not taken into account when bounds checking ?

**Comment by MarcoIeni** [comment]
I agree it's really complicated.
If someone wants to improve this a PR is welcome.

**Comment by pickfire** [comment]
I don't understand that sentence itself.

**Comment by kaifoon** [comment]
@pickfire I don't think so, it said method and trait are not taken into account. the statement link at below 
[Original Statement link](https://rust-unofficial.github.io/patterns/idioms/deref.html#disadvantages)

**Comment by kaifoon** [comment]
@MarcoIeni I think so too
